### ISSUE TITLE
ISSUE: Element already in chain of attachments. See console for more details.

### ISSUE THREAD

**Issue by krobibero** [KayelGee/token-attacher#93]
Issue with the latest version of Token Attacher.  Tested in both a Pathfinder and Starfinder worlds with the same results.

I receiving the following when trying to attach a token, to another token.  These tokens are not attached to anything else.

Element already in chain of attachments. See console for more details.
foundry.js:59187 

I've removed/reinstalled Token Attacher a few times which has not cured the issue.  I witnessed the folder for TokenAttacher be deleted and recreated to ensure it's actually being removed.  
Today I tried Find the Culprit!, which removed all modules accept Token Attacher and Lib-Wrapper.  The issue remains..

The console does not show any other details, just the message above.

I'm at a loss.

**Comment by krobibero** [comment]
Here is more details.  This is coming from within the console.

foundry.js:22349 
        
       Uncaught (in promise) TypeError: undefined. Cannot read properties of null (reading '_preUpdate')
[No packages detected]
    at SFRPGTokenDocument._preUpdateTokenActor (foundry.js:22349:24)
    at SFRPGTokenDocument._preUpdate (foundry.js:22262:18)
    at async ClientDatabaseBackend._preUpdateDocumentArray (foundry.js:12734:7)
    at async ClientDatabaseBackend._updateEmbeddedDocuments (foundry.js:12661:22)
_preUpdateTokenActor @ foundry.js:22349
_preUpdate @ foundry.js:22262

**Comment by KayelGee** [comment]
Sorry for not responding earlier.
I'm not exactly understanding what you're doing to cause the issue.
Are you using the manual attach method: https://github.com/KayelGee/token-attacher#attach-or-detach-element
Or are you using the multi layer select tool: https://github.com/KayelGee/token-attacher#attach-with-multi-layer-selection-tool
If you're using the first variant then you are not allowed to select both tokens when you click the attach button. You have to select the token you want to attach to the base token only and then click the attach button. I recommend using the multi layer selection tool, as it's usually more straightforward.

**Comment by krobibero** [comment]
I was tested some more first by using the Find the Culprit module.  Using it, I disabled all the modules except Token Attacher and lib-wrapper.  After going through that process, it found that no other module seemed to be causing the issue. 

Then I tried to recreate the issue to give you better perspective.

Testing Steps: 

2. Drop a token into a scene that does not have anything attached.  
3. Click the token, then click the Open Attaching UI button.  (Token Attacher UI appears)
4. Click the Attach Selected Elements button
That is when the message "Element already in chain of attachments. See console for more details."  pops-up.

In the console there is also
Uncaught (in promise) TypeError: undefined. Cannot read properties of null (reading '_preUpdate')
[No packages detected]
    at SFRPGTokenDocument._preUpdateTokenActor (foundry.js:22349:24)
    at SFRPGTokenDocument._preUpdate (foundry.js:22262:18)
    at async ClientDatabaseBackend._preUpdateDocumentArray (foundry.js:12734:7)
    at async ClientDatabaseBackend._updateEmbeddedDocuments (foundry.js:12661:22)

Does this help?
